bus/routes.go

```diff
@@ -955,6 +898,90 @@ func (b *Bus) contractIDHandlerPOST(jc jape.Context) {
 	jc.Encode(a)
 }
 
+func (b *Bus) contractIDRenewHandlerPOST(jc jape.Context) {
+	// apply pessimistic timeout
+	ctx, cancel := context.WithTimeout(jc.Request.Context(), 15*time.Minute)
+	defer cancel()
+
+	// decode contract id
+	var fcid types.FileContractID
+	if jc.DecodeParam("id", &fcid) != nil {
+		return
+	}
+
+	// decode request
+	var rrr api.ContractRenewRequest
+	if jc.Decode(&rrr) != nil {
+		return
+	}
+
+	// validate the request
+	if rrr.EndHeight == 0 {
+		http.Error(jc.ResponseWriter, "EndHeight can not be zero", http.StatusBadRequest)
+	} else if rrr.ExpectedNewStorage == 0 {
+		http.Error(jc.ResponseWriter, "ExpectedNewStorage can not be zero", http.StatusBadRequest)
+	} else if rrr.MaxFundAmount.IsZero() {
+		http.Error(jc.ResponseWriter, "MaxFundAmount can not be zero", http.StatusBadRequest)
+	} else if rrr.MinNewCollateral.IsZero() {
+		http.Error(jc.ResponseWriter, "MinNewCollateral can not be zero", http.StatusBadRequest)
+	} else if rrr.RenterFunds.IsZero() {
+		http.Error(jc.ResponseWriter, "RenterFunds can not be zero", http.StatusBadRequest)
+		return
+	}
+
+	// fetch the contract
+	c, err := b.ms.Contract(ctx, fcid)
+	if errors.Is(err, api.ErrContractNotFound) {
+		jc.Error(err, http.StatusNotFound)
+		return
+	} else if jc.Check("couldn't fetch contract", err) != nil {
+		return
+	}
+
+	// fetch the host
+	h, err := b.hs.Host(ctx, c.HostKey)
+	if jc.Check("couldn't fetch host", err) != nil {
+		return
+	}
+
+	// fetch consensus state
+	cs := b.cm.TipState()
+
+	// fetch gouging parameters
+	gp, err := b.gougingParams(ctx)
+	if jc.Check("could not get gouging parameters", err) != nil {
+		return
+	}
+
+	// send V2 transaction if we're passed the V2 hardfork allow height
+	var newRevision rhpv2.ContractRevision
+	var contractPrice, fundAmount types.Currency
+	if b.isPassedV2AllowHeight() {
+		panic("not implemented")
+	} else {
+		newRevision, contractPrice, fundAmount, err = b.renewContract(ctx, cs, gp, c, h.Settings, rrr.RenterFunds, rrr.MinNewCollateral, rrr.MaxFundAmount, rrr.EndHeight, rrr.ExpectedNewStorage)
+		if errors.Is(err, api.ErrMaxFundAmountExceeded) {
+			jc.Error(err, http.StatusBadRequest)
+			return
+		} else if jc.Check("couldn't renew contract", err) != nil {
+			return
+		}
+	}
+
+	// add renewal contract to store
+	r, err := b.addRenewedContract(ctx, fcid, newRevision, contractPrice, fundAmount, cs.Index.Height, api.ContractStatePending)
+	if jc.Check("couldn't store contract", err) != nil {
+		return
+	}
+
+	// send the response
+	jc.Encode(api.ContractRenewResponse{
```

Comment:
It would be nice to return `api.ContractMetadata` here. Perhaps we should move these fields (in some way shape or form) to the metadata (like we already do with `ContractPrice`) so both `form` and `renew` return contract metadata.

Reply:
The `NewCollateral` is only used for logging anyway right? The `FundAmount` is the `TotalCost` I think?

I also think there is a task on the v2 API issue regarding this. Because right now `TotalCost` is essentially equal to `FundAmount`. To make that clear I proposed renaming it to `InitialRenterFunds` throughout the codebase. 

But I'd double and triple check if that is ok to make sure we don't mess up some math anywhere

Reply:
F/U with the `InitialRenterFunds` thing for v2 API - I didn't want to get rid of the log context but def. not worth changing the return type for. Updated.